frontend/lib/resolution/call-init-deinit.cpp

```diff
@@ -616,7 +616,7 @@ void CallInitDeinit::resolveMoveInit(const AstNode* ast,
   if (isTypeParam(lhsType.kind())) {
     // OK, nothing else to do
   } else if (isValue(lhsType.kind()) && isValueOrParam(rhsType.kind())) {
-    if (lhsType.type() == rhsType.type()) {
+    if (canPass(context, rhsType, lhsType).passes()) {
```

Comment:
I think ``canPass`` is probably technically correct at the moment, but I'm a bit worried it might trigger for cases we didn't really intend to consider move-initialization. For example, initializing a ``real`` from an ``int``. Given that this guards an empty block it's probably not an issue for now though.

Maybe just a ``TODO`` asking whether ``canPass`` is too broad, and wondering if we should also be checking against ``canPass(...).converts()``?

Reply:
Good point. I've just updated it to accept a passing result that either doesn't require conversion, or only requires subtype conversion. Does that seem narrow enough?

Reply:
That sounds good to me. If it needs changing we'll figure it out during testing.